ui/components/app/confirm/info/info.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowProps,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowDivider,
+  ConfirmInfoRowValueDouble,
+  ConfirmInfoRowVariant,
+} from './row';
+import { ConfirmInfoContainer } from './container';
+
+export enum ConfirmInfoRowType {
+  Address = 'address',
+  Divider = 'divider',
+  ValueDouble = 'value-double',
+}
+
+export type ConfirmInfoRowConfig = {
+  /** The display label text. This should be required unless it is a 'divider' variant */
+  label?: ConfirmInfoRowProps['label'];
+
+  /** Optional, and likely needed, props passed to the row */
+  rowProps?: Record<string, any>;
+
+  /** The type of the row e.g. address, divider, value-double */
+  type: ConfirmInfoRowType;
+
+  /** Optional row variant  */
+  variant?: ConfirmInfoRowVariant;
+};
+
+interface ConfirmInfoProps {
+  rowConfigs: ConfirmInfoRowConfig[];
+}
+
+export const ConfirmInfo: React.FC<ConfirmInfoProps> = ({
+  rowConfigs = [],
+}) => (
+  <ConfirmInfoContainer>
+    {rowConfigs.map((rowConfig: ConfirmInfoRowConfig, index) => {
+      const { label, rowProps, type, variant } = rowConfig;
+      const key = `confirm-info-row-${label}-${index}`;
+
+      switch (type) {
+        case ConfirmInfoRowType.Address:
+          return (
+            <React.Fragment key={key}>
+              <ConfirmInfoRow label={label || ''} variant={variant}>
+                <ConfirmInfoRowAddress address={rowProps?.address} />
+              </ConfirmInfoRow>
+            </React.Fragment>
+          );
+        case ConfirmInfoRowType.Divider: {
+          const dividerKey = `confirm-info-divider-${rowConfigs
+            .map(({ label: _label }) => _label)
+            .concat('-')}-${index}`;
+
+          return (
+            <React.Fragment key={dividerKey}>
+              <ConfirmInfoRowDivider />
+            </React.Fragment>
+          );
+        }
+        case ConfirmInfoRowType.ValueDouble:
+          return (
+            <React.Fragment key={key}>
+              <ConfirmInfoRow label={label || ''} variant={variant}>
+                <ConfirmInfoRowValueDouble
+                  left={rowProps?.left}
+                  right={rowProps?.right}
+                />
+              </ConfirmInfoRow>
+            </React.Fragment>
+          );
+        default:
+          return null;
+      }
+    })}
+  </ConfirmInfoContainer>
+);
```

Comment:
for documentation purposes. Initially, I thought we were handling the logic here, but then it was discussed the parent will specify which rows to display through params.

Initial proposal 
- no params passed, or transaction data object 
- no need for a divider component. specifying an array of an array would support displaying sections divided by divider (see POC of [sections here](https://github.com/MetaMask/metamask-extension/pull/22198)) 

example POC (non-executable)
```
// Map transaction type to translation keys
const typeToInfoKeyMap: Partial<{ [key in TransactionType]: string[][] }> = {
  [TransactionType.personalSign]: ['origin', 'divider', 'message'],
  // [TransactionType.personalSign]: [['origin'], ['message']],
  // [TransactionType.personalSign]: ['origin', 'method', 'divider' , 'message', 'uri', 'issued'],
};


export const ConfirmInfo: React.FC = ({}) => {
  const currentConfirmation = useSelector(currentConfirmationSelector);

  if (!currentConfirmation) {
    return '';
  }

  // there is no 1-to-1 mapping from transaction type to display rows. This logic will get more complicated as we support more pages 
  const infoKeys = typeToInfoKeyMap[currentConfirmation.type];

  return (
    <>
      {infoKeys?.map((infoBox) => {
        const infoSections = infoBox.split('divider');

        return (
          <ConfirmInfoBox key={infoSections.join('')}>
            {infoSections.map((infoSectionKeys) => {
              return (
                <InfoBoxSection
                  key={infoKey}
                  infoKey={infoKey}
                  value={currentConfirmation[infoKey]}
                >
                  {infoSectionKeys.map((infoKey) => {
                    return (
                      // get row by info key and pass value
                    );
                  })}
                </InfoBoxSection>
              );
            })}
          </ConfirmInfoBox>
        );
      })}
    </>
  );
};
```